src/service/data_products/genome_attributes.py

```diff
@@ -653,6 +653,13 @@ async def _mark_gtdb_matches_STARTS_WITH_strategy(
     # later
     # could also probably DRY up this and the above method
     # retries?
+    if not lineages:
+        # return no matches
+        await storage.update_match_state(
+            internal_match_id, models.ProcessState.COMPLETE, now_epoch_millis(), []
+        )
+        return
```

Comment:
You should do this in `perform_gtdb_lineage_match`, otherwise you'll miss some cases since the matcher chooses one of the two *strategy functions

Reply:
also maybe make this a function and use that function wherever the same code is called

Reply:
👍 I moved it to `perform_gtdb_lineage_match`. It's just calling `storage.update_match_state`. Making this single function call a function?

Reply:
Fair enough, I guess you'd only be getting rid of a couple args, and the millis function probably shouldn't be abstracted out anyway since it's not mockable and needs to be changed

Reply:
I guess it is mockable by monkey patching but I hate that